src/NexusMods.MnemonicDB/Transaction.cs

```diff
@@ -150,39 +162,96 @@ public bool TryGet<TEntity>(EntityId entityId, [NotNullWhen(true)] out TEntity?
         return false;
     }
 
+    public void CommitToParent()
+    {
+        Debug.Assert(_parentTransaction is not null);
+
+        var indexSegment = _datoms.Build();
+        foreach (var datom in indexSegment)
+        {
+            _parentTransaction.Add(datom);
+        }
```

Comment:
Should `_datoms` also be cleared?

If you commit a sub transaction, should it be possible to continue using it adding more changes to it to commit to parent a second time or should it get disposed after the first commit to parent?

The main root transaction can only be committed once, should child transactions have the same semantics?

Reply:
Updated so `_committed` gets set the true for sub-transactions. Cleanup will be done in the `Reset` method.